### DISCUSSION TITLE
is there an "interface shells" option in Orca Slicer?

### DISCUSSION THREAD

**Discussion by CyberManiaK** [OrcaSlicer/OrcaSlicer#5106] [Q&A]
Anyone know where is or how is achieved the "interface shells" option that prusaslicer has? This screenshot is from PrusaSlicer, what it does, is that when using multimaterial (multicolor) printing it puts a solid top layer before the next object, which in this case is a text, I have added a screenshot, so you can better understand what I mean. The top one is Orcaslicer same layer height, same # of bottom/top layers,etc, and the bottom one is Prusa, and as you can see just with the option of interface shells ticked it doesn't make the cuts for the text, so instead we have a nice top solid layer that is waiting for the text to be printed. Thanks. 
![20240221_140634](https://github.com/SoftFever/OrcaSlicer/assets/952352/bf7bc0c2-59ba-4775-847e-d8e49f56196c)
![Screenshot 2024-02-21 135805](https://github.com/SoftFever/OrcaSlicer/assets/952352/7407c06f-beb1-4057-8057-17ea96311785)

**Comment by CyberManiaK** [comment]
Bump!! I guess this is not available in Orca, can this be requested then? It's super helpfull for users doing multicolor with a single extruder.

**Comment by TrueScott** [comment]
Oh, I just asked same question, I will delet mine!
Bambu Studio has this feature under dev mode, but not Orca Slicer...
I think that's a great option!

**Comment by NVNDO** [comment]
Hey @CyberManiaK just saw your post on Reddit when I searched for a solution on Orca. 
Me and a bunch of people fought a long time to get this implemented in Bambu Studio. They also call it "Interface shells".
https://github.com/bambulab/BambuStudio/issues/2906

To make this work in Bambu Studio - you activate "Interface shells".. ..and "Topmost surface"
![mycreation23 01 2025_14_12_49](https://github.com/user-attachments/assets/ee308da5-d54f-40f8-85ec-e17c9bd1b174)

@SoftFever  *WE NEED THIS IN ORCA AS WELL!*

Due to not having this option, Orca is no longer an option for me (and probably many others)
**Please implement this because Orca is generally such a great Slicer!!**

**Comment by SoftFever** [comment]
The shell interfaces option has been exposed now. Ref to PR: #8411
This is a PrusaSlicer feature but was hidden from UI previous in Orca.
It has now been espoused.
Please pay attention that it works best with "one perimeter on top surface" off based on my primary test.
To make it work problem, you need to paint to model, add filament change custom g-code won't work for this feature.
If you need manual filament change, please use the Manual Filament Change option from Orca.
Which will allows you manual change filament with painting.
<img width="364" alt="Screenshot 2025-02-15 at 2 42 13 PM" src="https://github.com/user-attachments/assets/b86feeb2-6e99-4143-86f4-6a521b33bbca" />

[KayTag_Test.3mf.zip](https://github.com/user-attachments/files/18809001/KayTag_Test.3mf.zip)

**Reply by jeremytodd1** [reply]
This is **so** close to what I've been needing for close to two years now.
The fact that I can get the layer below the text to print cleanly is amazing, but having to disable single perimeters on top layers is another deal breaker for my use case unfortunately.

Any ideas on if this feature will work with single perims on top layer in the future?

This has been quite an issue for me over the years, as 95% of the prints I sell are affected by this issue. I'm very excited that Orca is starting to have a solution for this issue.

**Reply by SoftFever** [reply]
Yeah, it seems a bug in current single wall implementation.
I plan to investigate that issue.

**Reply by SoftFever** [reply]
It indeed sucks when a print has to be made without 'One wall on the top surface'! ;)
This issue should be addressed in this change: https://github.com/SoftFever/OrcaSlicer/pull/8627
The enhancement will be included in the upcoming 2.3.0 beta2 release. However, feel free to try the PR build and let me know if you encounter any issues.

**Reply by jeremytodd1** [reply]
Amazing, I'm very glad to hear that, thank you so much.
I do have one last question, if you don't mind.
I print a ton of 3d maps. I'd say maps are 90% of what I print, and the other 10% are random printer parts and whatnot.
This issue is quite a big one for me.

Here is a quick test example I made to help show my issue. 
![pic1](https://github.com/user-attachments/assets/6da207e8-e9a9-4980-89ab-9aeb58262bba)

The maps I make all have a base that is 4 layers thick. The first two layers are black and then I switch to white. I do this because it helps give the map some contrast so it's easier to see the roads, otherwise it's too white. All of the map details (buildings/roads) ideally will *start* printing on layer 5.
So the first few layers should look like this:
Layer 1: The base of the map starts printing inblack filament
Layer 2: black filament
Pause print and switch to white filament
Layer 3: white filament
Layer 4: white filament
Layer 5: This layer is when all the map details should start printing. 

With this solution, it only really works when the filament change is on the same layer as the top details. 
It seems like I'd still be having issues in my scenario after this solution, since my filament change is on a different layer.
Sorry if I worded this oddly. It's kind of tough to describe easily.

**Reply by SoftFever** [reply]
do you have a sample project file? I feel you probably can use the height range in color painting tool?

**Reply by jeremytodd1** [reply]
[interface_layers_testing.zip](https://github.com/user-attachments/files/19045260/interface_layers_testing.zip)

Just made a quick test for it. (sorry, I had to zip it in order to attach it. If you're not comfortable with that I can upload it to printables instead)

I'm trying this on the latest beta2 release that came out earlier today.

It's just a small tile of a map. 
The base of the map is .8mm thick. I print at a .2 layer height so it's 4 layers thick.
The first 2 layers are black, then it switches to white for the remainder of the print.
The biggest thing is that layer 4 needs to print clean. There should be no building/road details until layer 5. 
One wall on top layers is needed too, as it looks much nicer.

I've been playing around with it for a while today and haven't gotten it to slice ideally still. If you manage to find settings that will work for this please let me know, as that would simplify my processes immensely for making these maps.

**Comment by NVNDO** [comment]
Could someone please explain how this works... because I'm trying since an hour without any success...

"Interface shells" does not show in "Multimaterial", regardless of what I do/ change. I'm on the latest version 2.2.0

I assume "one perimeter on top surface" means, Wall loops? Top Layer? 
I set those to one, but the option does not show.

Please help 🙏🏼

**Reply by henrivdr** [reply]
The feature was only Merged in the PR mentioned above. So it won't be in 2.2.0
Have a look at the latest OrcaSlicer v2.3.0-beta Release
https://github.com/SoftFever/OrcaSlicer/releases

**Reply by NVNDO** [reply]
Oh damn... sorry. I just looked at the current release and the beta did not show - therefore I though being on the most recent release. Thx a lot for the link - Though I still don't get it to work. I'm back to Bambu Studio... I'm tired. 2 hours trying to figure something out because its not sufficiently explained. Grrr....

**Comment by DanielSchenk1337** [comment]
Awesome! Multicolour prints look way better with interface shells! 

One Question: is it possible to somehow enable this feature for singe material prints? This would enable much stronger connections between features in some cases, at least i think so.

My workaround for now is to colour features with two colours and select one filament to print both, it works but is quite wasteful.

**Reply by SoftFever** [reply]
It's a bit tricky as the the interface_shells rely on multi material painting
As a workaround, you can achieve the same result by applying a modifier tool to set the fill density of that area to 100%.

**Comment by henrivdr** [comment]
@DanielSchenk1337 I agree.  I see it is under Strength settings in Bambu.
I use the same workaround by designing the features (which end up with weak bonds to a lower surface) as separate parts.
Then set them with the same filament but different color.

Weak. Only bonded by it's single wall.
![image](https://github.com/user-attachments/assets/12cad442-6beb-4f6a-8460-fcf107f861fa)


Strong with Interface Shell. Full surface bond.
![image](https://github.com/user-attachments/assets/1f3eaff2-fed5-42c8-ba16-93a722360737)

**Comment by DanielSchenk1337** [comment]
The yellow lines on layer 6 are just printed in free air more or less... Interface shells for the entire model would solve this.
![RUIDe60f35b9e43b49a3b9b3fd503d521e7c](https://github.com/user-attachments/assets/71ef27f2-cb1a-406f-b06b-7a5704664839)
Layer 5
![Screenshot (41)](https://github.com/user-attachments/assets/809c13f8-dc90-44cd-97de-6dbf6c042c38)
Layer 6
![Screenshot (42)](https://github.com/user-attachments/assets/2ecae32d-d527-40bc-b593-85d89d1e36b7)
